src/test/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByStateUnitTest.java

```diff
@@ -0,0 +1,745 @@
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.utils.NGSPlatform;
+import org.broadinstitute.hellbender.utils.QualityUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.downsampling.DownsampleType;
+import org.broadinstitute.hellbender.utils.downsampling.DownsamplingMethod;
+import org.broadinstitute.hellbender.utils.pileup.PileupElement;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.read.ArtificialBAMBuilder;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+public final class LocusIteratorByStateUnitTest extends LocusIteratorByStateBaseTest {
+    protected LocusIteratorByState li;
+
+    @Test
+    public void testIterator() {
+        final int readLength = 10;
+        final GATKRead mapped1 = ArtificialReadUtils.createArtificialRead(header, "mapped1", 0, 1, readLength);
+        final GATKRead mapped2 = ArtificialReadUtils.createArtificialRead(header, "mapped2", 0, 1, readLength);
+
+        final List<GATKRead> reads = Arrays.asList(mapped1, mapped2);
+
+        li = makeLIBS(reads, DownsamplingMethod.NONE, true, header);
+        Assert.assertSame(li.iterator(), li);
+    }
+
+    @Test(expectedExceptions = NoSuchElementException.class)
+    public void testIteratingBeyondElements() {
+        final int readLength = 3;
+        final GATKRead mapped1 = ArtificialReadUtils.createArtificialRead(header, "mapped1", 0, 1, readLength);
+
+        final List<GATKRead> reads = Arrays.asList(mapped1);
+
+        li = makeLIBS(reads, DownsamplingMethod.NONE, true, header);
+        for (int i = 0; i < readLength; i++) {
+            Assert.assertTrue(li.hasNext());
+            Assert.assertNotNull(li.next());
+        }
+        Assert.assertFalse(li.hasNext());
+        li.next();
+    }
+
+    @Test
+    public void testAdvance() {
+        final int readLength = "ACTG".length();
+        final GATKRead mapped1 = ArtificialReadUtils.createArtificialRead(header, "mapped1", 0, 1, readLength);
+        mapped1.setBases("ACTG".getBytes());
+
+        final List<GATKRead> reads = Arrays.asList(mapped1);
+
+        li = makeLIBS(reads, DownsamplingMethod.NONE, true, header);
+        final AlignmentContext alignmentContext2 = li.advanceToLocus(2, true);
+        Assert.assertEquals(alignmentContext2.getPosition(), 2);
+
+        final AlignmentContext alignmentContext3 = li.advanceToLocus(3, true);
+        Assert.assertEquals(alignmentContext3.getPosition(), 3);
+
+        final AlignmentContext alignmentContext10 = li.advanceToLocus(10, true);
+        Assert.assertNull(alignmentContext10);
+
+    }
+
+    @Test(enabled = true)
+    public void testAllIReadsPassThrough() {
```

Comment:
What about unmapped reads? There used to be a test that unmapped reads pass through as well -- has the contract changed in the ported version?